blockmanager.go

```diff
@@ -3043,4 +3097,177 @@ func (l *lightHeaderCtx) RelativeAncestorCtx(
+	err := b.sideLoadReader.SetHeight(int64(b.headerTip))
+	if err != nil {
+		log.Errorf("error while setting height for sideload--- skipping sideloading: "+
+			"%v", err)
+
+		return
+	}
+
+	// Start batch writer to write headers into DB at specified batch size and ticker duration.
+	b.blockHeaderBatchWriter.Start()
+	defer b.blockHeaderBatchWriter.Stop()
+	for {
+		select {
+		case <-b.quit:
+			return
+		default:
+			//Request header
+			header, headerErr := b.sideLoadReader.NextHeader()
+			// If any error occurs while fetching headers that does not indicate
+			// an end of file, return immediately.
+			if headerErr != nil {
+
+				if headerErr == chainDataLoader.ErrEndOfFile {
+					log.Infof("Successfully completed sideloading")
+					return
+				}
+
+				log.Errorf("error while fetching headers -- skipping sideloading %v", err)
+				return
+			}
+
+			var (
+				node     *headerlist.Node
+				prevNode *headerlist.Node
+			)
+			// Ensure there is a previous header to compare against.
+			prevNodeEl := b.headerList.Back()
+			if prevNodeEl == nil {
+				log.Warnf("side load - Header list does not contain a previous" +
+					"element as expected -- exiting side load")
+
+				return
+			}
+
+			node = &headerlist.Node{Header: *header}
+			prevNode = prevNodeEl
+			node.Height = prevNode.Height + 1
+
+			if b.cfg.sideLoad.Verify {
+				valid, err := b.verifyBlockHeader(header, *prevNode)
+				if err != nil || !valid {
+					log.Debugf("Side Load- Did not pass verification at height %v"+
+						"-- rolling back to last verified checkpoint and skipping sideload", node.Height)
+
+					prevCheckpoint := b.findPreviousHeaderCheckpoint(
+						node.Height,
+					)
+
+					log.Infof("Rolling back to previous validated "+
+						"checkpoint at height %d/hash %s",
+						prevCheckpoint.Height,
+						prevCheckpoint.Hash)
+
+					//It is possible that the number of headers in the batch writer's buffer is less than the
+					//BlkHeaderWriteBatch and so these headers have not been written into the DB at this time.
+					//We wait for the DBTickerBlkHeaderWriteDuration to expire and write these headers into the
+					//DB before we roll back which should take approximately twice the DBTickerBlkHeaderWriteDuration.
+					//This ensures that these headers are wiped off, during the rollback, in the batch writer's buffer
+					//and in the DB as well.
+					time.Sleep(DBTickerBlkHeaderWriteDuration * 2)
+					err = b.rollBackToHeight(uint32(prevCheckpoint.Height))
+					if err != nil {
+						panic(fmt.Sprintf("Rollback failed: %s", err))
+						// Should we panic here?
+					}
+					tipHeader, height, err := b.cfg.BlockHeaders.ChainTip()
+					if err != nil {
+						return
+					}
+
+					b.headerList.ResetHeaderState(headerlist.Node{
+						Height: int32(height),
+						Header: *tipHeader,
+					})
+					return
+				}
+
+			}
+
+			// Verify checkpoint only if verification is enabled.
+			if b.nextCheckpoint != nil && b.cfg.sideLoad.Verify &&
```

Comment:
`if b.nextCheckpoint != nil && b.cfg.sideLoad.Verify && node.Height == b.nextCheckpoint.Height {`

The `if` in top of this one is already checking `if b.cfg.sideLoad.Verify {}` maybe we can put the check inside and remove the double check for the `b.cfg.sideLoad.Verify`?

Reply:
Makes sense. Thanks,